tests/optimize/curve_fit_test.py

```diff
@@ -0,0 +1,119 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+import numpy as np
+import scipp as sc
+from scipp.optimize import curve_fit
+
+import pytest
+
+
+def func(x, a, b):
+    return a * sc.exp(-(b / x.unit) * x)
+
+
+def array1d(*, a=1.2, b=1.3, noise_scale=0.1):
+    size = 50
+    x = sc.linspace(dim='xx', start=-0.1, stop=4.0, num=size, unit='m')
+    y = func(x, a, b)
+    rng = np.random.default_rng()
+    # Noise is random but avoiding unbounded values to avoid flaky tests
+    y.values += noise_scale * np.clip(rng.normal(size=size), -2.0, 2.0)
+    return sc.DataArray(y, coords={'xx': x})
+
+
+def test_should_not_raise_given_function_with_dimensionless_params_and_1d_array():
```

Comment:
This is covered by other test cases. Is it supposed to check `func` and `array1d` instead of `curve_fit`?

Reply:
It was meant to validate that the `ValueError` I get in the 3 tests below comes indeed from the bad xdata/ydata/sigma arguments and not some other bad argument. I had originally put this assert into those tests, but that was also odd.